asyncpgsa/connection.py

```diff
@@ -15,32 +15,41 @@
 _dialect.paramstyle = 'named'
 
 
+def replace_keys(keys, querystring, params, inline=False):
+    new_query = querystring
+    new_params = []
+    for key, value in enumerate(keys):
+        if inline:
+            new_query = new_query.replace(':' + value, params[key][1], 1)
+        else:
+            new_query = new_query.replace(':' + value, '$' + str(key + 1), 1)
+            new_params.append(params[key][1])
+
+    return new_query, new_params
+
+
+def get_keys(compiled):
+    import re
+    pattern = ':(\w+)(?:\W|)'
```

Comment:
rexex string should be prepended with a `r`. `r'mypattern'`

Also, we can improve time a little bit by compiling the pattern at the top of the file rather than every time in this function.